test/e2e/minscale_readiness_test.go

```diff
@@ -150,3 +158,19 @@ func waitForDesiredScale(t *testing.T, clients *test.Clients, deploymentName str
 		return cond(*deployment.Spec.Replicas), nil
 	})
 }
+
+func waitForDesiredEndpoints(t *testing.T, clients *test.Clients, endpointName string, cond func(int32) bool) error {
+	endpoints := clients.KubeClient.Kube.CoreV1().Endpoints(test.ServingNamespace)
+
+	return wait.PollImmediate(time.Second, 10*time.Minute, func() (bool, error) {
+		endpoint, err := endpoints.Get(endpointName, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+
+		if len(endpoint.Subsets) == 0 || len(endpoint.Subsets[0].Addresses) == 0 {
+			return false, nil
+		}
+		return cond(int32(len(endpoint.Subsets[0].Addresses))), nil
```

Comment:
I think this should use the `resources.ReadyAddressCount` helper. See https://github.com/knative/serving/blob/afbd55a1df2d1548f603bd249b4672141beb8007/test/e2e/autoscale_test.go#L283-L302 for an example.

Reply:
Thanks for reviewing. I will improve it.